src/main/java/org/generationcp/middleware/manager/ontology/OntologyVariableDataManagerImpl.java

```diff
@@ -1051,4 +1101,76 @@ public long countVariablesByDatasetId(final Integer datasetId, final List<Intege
 	public long countAllVariables(final List<Integer> variableTypes) {
 		return this.daoFactory.getCvTermDao().countAllVariables(variableTypes);
 	}
+
+	@Override
+	public List<Variable> searchAttributeVariables(final String query, final String programUUID) {
+		if (StringUtils.isBlank(query)) {
+			return Collections.EMPTY_LIST;
+		}
+		try {
+			final SQLQuery sqlQuery = this.getActiveSession().createSQLQuery("select "
+				+ "v.cvterm_id vid, v.name vn, v.definition vd, vmr.mid, vmr.mn, vmr.md, vpr.pid, vpr.pn, vpr.pd, vsr.sid, vsr.sn, vsr.sd, \n"
+				+ "vpo.alias p_alias, vpo.expected_min p_min_value, vpo.expected_max p_max_value" //
+				+ " FROM cvterm v INNER JOIN cvtermprop cp ON cp.type_id = " + TermId.VARIABLE_TYPE.getId()
+				+ " and v.cvterm_id = cp.cvterm_id " //
+				+ " left join (select mr.subject_id vid, m.cvterm_id mid, m.name mn, m.definition md from cvterm_relationship mr inner join cvterm m on m.cvterm_id = mr.object_id and mr.type_id = 1210) vmr on vmr.vid = v.cvterm_id "
+				+ " left join (select pr.subject_id vid, p.cvterm_id pid, p.name pn, p.definition pd from cvterm_relationship pr inner join cvterm p on p.cvterm_id = pr.object_id and pr.type_id = 1200) vpr on vpr.vid = v.cvterm_id "
+				+ " left join (select sr.subject_id vid, s.cvterm_id sid, s.name sn, s.definition sd from cvterm_relationship sr inner join cvterm s on s.cvterm_id = sr.object_id and sr.type_id = 1220) vsr on vsr.vid = v.cvterm_id "
+				+ " left join variable_overrides vpo ON vpo.cvterm_id = v.cvterm_id AND vpo.program_uuid = :programUUID " //
+				+ " WHERE v.cv_id = " + CvId.VARIABLES.getId() + " "  //
+				+ "		and cp.value in (select name from cvterm where cvterm_id in (" //
+				+ VariableType.GERMPLASM_PASSPORT.getId() + "," //
+				+ VariableType.GERMPLASM_ATTRIBUTE.getId() + ")) " //
+				+ "   AND (v.definition like :fname or v.name like :name or vpo.alias like :alias )" //
+				+ " LIMIT 100 ");
+			sqlQuery.setParameter("programUUID", programUUID);
+			sqlQuery.setParameter("fname", '%' + query + '%');
+			sqlQuery.setParameter("name", '%' + query + '%');
+			sqlQuery.setParameter("alias", '%' + query + '%');
+			sqlQuery.setResultTransformer(Transformers.ALIAS_TO_ENTITY_MAP);
+			sqlQuery.setResultTransformer(Transformers.ALIAS_TO_ENTITY_MAP);
+			sqlQuery.addScalar(OntologyVariableDataManagerImpl.VARIABLE_ID);
```

Comment:
@cuenyad 
If you are using constants to refer to the scalars, then you should use constansts to build the query